src/server.ts

```diff
@@ -749,3 +753,77 @@ function findCreatedAccountSequenceInTransactionMeta(
 
   throw new Error("No account created in transaction");
 }
+
+
+// TODO - remove once rpc updated to
+// append expiration entry per data LK requested onto server-side response 
+// https://github.com/stellar/soroban-tools/issues/1010
+async function mergeResponseExpirationLedgers(response: Promise<SorobanRpc.RawGetLedgerEntriesResponse>,
+  requestedKeys: xdr.LedgerKey[]
+): Promise<SorobanRpc.RawGetLedgerEntriesResponse> {
+  let ledgerEntriesResponse = await response;
+  const expirationKeyToRawEntryResult:Map<String, SorobanRpc.RawLedgerEntryResult> = new Map();
```

Comment:
I think it'd be better to just use the response directly rather than the promise and put the `await` in the higher level (i.e. in `Server.getLedgerEntries`) - then this can be used & tested synchronously! 

```suggestion
function mergeResponseExpirationLedgers(
  response: SorobanRpc.RawGetLedgerEntriesResponse,
  requestedKeys: xdr.LedgerKey[]
): SorobanRpc.RawGetLedgerEntriesResponse {
  const ledgerEntriesResponse = response;
  const expirationKeyToRawEntryResult:Map<String, SorobanRpc.RawLedgerEntryResult> = new Map();
```

Reply:
https://github.com/stellar/js-soroban-client/pull/161/commits/e6e944a48b4a8bee1d7edc30e16ffb382d29be56